pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -111,14 +218,47 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 	templateData.ExternalStaticDNS = config.BootstrapExternalStaticDNS
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
-	_, externalURLv6 := externalURLs(config.APIVIPs)
+	// If the user has manually set disableVirtualMediaTLS to False in the Provisioning CR, then enable TLS in ironic.
+	// The default value of 'true' is temporary (for backporting to older releases), which will be changed to 'false' in the future.
+	templateData.DisableIronicVirtualMediaTLS = true
+	protocol := "http"
 
+	type provisioningCRTemplate struct {
+		Spec struct {
+			DisableVirtualMediaTLS *bool `yaml:"disableVirtualMediaTLS"`
+		} `yaml:"spec"`
+	}
+	provisioningCR := &provisioningCRTemplate{}
+	openshiftManifests := &manifests.Openshift{}
+	dependencies.Get(openshiftManifests)
+	var provisioningCRBytes []byte
+	for _, file := range openshiftManifests.Files() {
+		if strings.Contains(file.Filename, "99_baremetal-provisioning-config.yaml") {
+			logrus.Debugf("Provisioning CR found")
+			provisioningCRBytes = file.Data
+			break
+		}
+	}
+	if provisioningCRBytes != nil {
+		err := yaml.Unmarshal(provisioningCRBytes, provisioningCR)
+		if err != nil {
+			logrus.Debugf("Error in unmarshalling Provisioning CR while generating TLS certificate for ironic virtual media: %s", err)
+		}
+	}
```

Comment:
I think we should log something angry when provisioningCRBytes is nil

Reply:
Done